### PR TITLE
Filter for efs lifecycle-policy state

### PR DESCRIPTION
**Pull request by PratMis**
Follow-up PR of #5275 . This adds a filter to check the state of lifecycle-policies in efs.
Sample policy:
```
policies:
  - name: filter-efs-lifecycle
     resource: efs
     filters:
       - type: lifecycle-policy
          state: present
          value: AFTER_7_DAYS
```

### CODE REVIEW

File: c7n/resources/efs.py [hunk 1]
```diff
@@ -193,3 +194,42 @@ def process(self, resources):
...
+            policies:
+              - name: efs-filter-lifecycle
+                resource: efs
+                filters:
+                  - type: lifecycle-policy-enabled
+                    state: true
+
+    """
+    schema = type_schema(
+        'lifecycle-policy-enabled',
```
Comment (kapilt) [inline c7n/resources/efs.py]: perhaps drop the enabled suffix since we're doing the state boolean

Comment (PratMis) [reply c7n/resources/efs.py]: Sounds good!

File: c7n/resources/efs.py [hunk 2]
```diff
@@ -193,3 +194,42 @@ def process(self, resources):
...
+    def process(self, resources, event=None):
+        client = local_session(self.manager.session_factory).client('efs')
+        results = []
+        for r in resources:
+            try:
+                response = client.describe_lifecycle_configuration(
+                    FileSystemId=r['FileSystemId'])
+            except client.exceptions.FileSystemNotFound:
+                continue
+            if self.data.get('state') and response.get('LifecyclePolicies'):
```
Comment (kapilt) [inline c7n/resources/efs.py]: I feel like we should go ahead add in the ability to check on the actual policy content with this filter, ie is it 7 days, 14 etc. The lifecycle policies are an array but only support a singleton value at the moment, so from a filter perspective perhaps ie `value: AFTER_7` with `state: present` to only find those configured for 7 days.

Comment (PratMis) [reply c7n/resources/efs.py]: Cool. Was thinking more of making `value` an optional parameter that would be of type array.  Although, as you said, it only supports a single value right now but might be worth keeping it as an array for future. Thoughts? Thanks

Comment (kapilt) [reply c7n/resources/efs.py]: I think we can do it as singleton non array, and if they expose more options later we can extend it w/ compatibility.

Comment (PratMis) [reply c7n/resources/efs.py]: Cool, thanks! Also realized we can manipulate `value` to attain any possible filter. So I made it a mandatory field to keep things simple.

File: c7n/resources/efs.py [hunk 3]
```diff
@@ -193,3 +194,45 @@ def process(self, resources):
...
+                  - type: lifecycle-policy
+                    state: present
+                    value: AFTER_7_DAYS
+
+    """
+    schema = type_schema(
+        'lifecycle-policy',
+        state={'enum': ['present', 'absent']},
+        value={'type': 'string'},
+        required=['state', 'value'])
```
Comment (kapilt) [inline c7n/resources/efs.py]: I think its valid to look for state: present or state: absent sans specifying a value here.

Comment (PratMis) [reply c7n/resources/efs.py]: Thanks, addressed this

File: c7n/resources/efs.py [hunk 4]
```diff
@@ -193,3 +194,52 @@ def process(self, resources):
...
+                return [r for r in resources if r.get('LifecyclePolicies')]
+            return [r for r in resources if r.get('LifecyclePolicies') == []]
+
+    def lfc_evaluation(self, resources):
+        client = local_session(self.manager.session_factory).client('efs')
+        for r in resources:
+            try:
+                lfc = client.describe_lifecycle_configuration(
+                    FileSystemId=r['FileSystemId']).get('LifecyclePolicies')
+                r['LifecyclePolicies'] = lfc
```
Comment (kapilt) [inline c7n/resources/efs.py]: annotations needs a c7n: prefix

File: c7n/resources/efs.py [hunk 5]
```diff
@@ -193,3 +194,52 @@ def process(self, resources):
...
+            config = {'TransitionToIA': self.data.get('value')}
+            if self.data.get('state') == 'present':
+                return [r for r in resources if config in r.get('LifecyclePolicies')]
+            return [r for r in resources if config not in r.get('LifecyclePolicies')]
+        else:
+            if self.data.get('state') == 'present':
+                return [r for r in resources if r.get('LifecyclePolicies')]
+            return [r for r in resources if r.get('LifecyclePolicies') == []]
+
+    def lfc_evaluation(self, resources):
```
Comment (kapilt) [inline c7n/resources/efs.py]: the method name here could use a different name, its not really evaluating, its just fetching/annotating lifecycle

**Comment by PratMis** [comment]
Thanks @kapilt ! Addressed the findings as per your suggestion

**Pull request merged**